src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseDownloader.java

```diff
@@ -179,24 +170,29 @@ public String formatPath(String path) {
     }
 
     /**
-     * Get advantage percent in downloading single exercise.
-     *
-     * @param exCount order number of exercise in downloading
-     * @param exercisesSize total amount of exercises that will be downloaded
-     * @return percents
-     */
-    public double getPercents(int exCount, int exercisesSize) {
-        return Math.round(1.0 * exCount / exercisesSize * 100);
-    }
-
-    /**
      * Downloads single .zip file by using URLCommunicator.
      *
      * @param zipUrl url which will be downloaded
      * @param path where to download
      */
-    private void downloadExerciseZip(URI zipUrl, String path) {
+    private boolean downloadExerciseZip(URI zipUrl, String path) {
         File file = new File(path);
-        urlCommunicator.downloadToFile(zipUrl, file);
+        return urlCommunicator.downloadToFile(zipUrl, file);
+    }
+
+    private class SingleExerciseDownloadHandler implements Callable<Boolean> {
```

Comment:
Well it calls non-static method handleSingleExercise which calls other non-static methods. I could take Exersice downloader as parameter of constructor to get around that, but that wouldn't really make difference.